ext/mysqli/tests/test_setup/test_helpers.inc

```diff
@@ -0,0 +1,164 @@
+        $password = getenv('MYSQL_TEST_PASSWD') ?: '';
+    }
+    return $password;
+}
+function get_default_database(): string {
+    static $db = null;
+    if ($db === null) {
+        $db = getenv('MYSQL_TEST_DB') ?: 'test';
+    }
+    return $db;
+}
+function get_default_db_engine(): string {
+    static $engine = null;
+    if ($engine === null) {
+        $engine = getenv('MYSQL_TEST_ENGINE') ?: 'InnoDB';
+    }
+    return $engine;
+}
+function get_default_socket(): ?string {
+    /*
+    static $socket = null;
+    if ($socket === null) {
+        $socket = getenv('MYSQL_TEST_ENGINE') ?: null;
+        if ($socket) {
+            ini_set('mysqli.default_socket', $socket);
+        }
+    }
+    return $socket;
+    */
+    return null;
+}
+function get_environment_connection_flags(): int {
+    static $connect_flags = null;
+    if ($connect_flags === null) {
+        $connect_flags = (int)getenv("MYSQL_TEST_CONNECT_FLAGS") ?: 0;
+    }
+    return $connect_flags;
+}
+
+/**
+ * Whenever possible, please use this wrapper to make testing of MYSQLI_CLIENT_COMPRESS (and potentially SSL) possible
+ *
+ * @param bool $enable_env_flags Enable setting of connection flags through 	env(MYSQL_TEST_CONNECT_FLAGS)?
+ */
+function my_mysqli_connect(
+    string $host,
+    string $user,
+    string $password,
+    string $db,
+    int $port,
+    ?string $socket = null,
+    bool $enable_env_flags = true
+): \mysqli|false {
+
+    $flags = $enable_env_flags? get_environment_connection_flags():0;
+    if ($flags !== 0) {
+        $link = mysqli_init();
+        if (!mysqli_real_connect($link, $host, $user, $password, $db, $port, $socket, $flags)) {
+            $link = false;
+        }
+    } else {
+        $link = mysqli_connect($host, $user, $password, $db, $port, $socket);
+    }
+
+    return $link;
+}
+
+/**
+ * Whenever possible, please use this wrapper to make testing of MYSQLI_CLIENT_COMPRESS (and potentially SSL) possible
+ *
+ * @param bool $enable_env_flags Enable setting of connection flags through env(MYSQL_TEST_CONNECT_FLAGS)
+ */
+function my_mysqli_real_connect(
+    mysqli $link,
+    string $host,
+    string $user,
+    string $password,
+    string $db,
+    int $port,
+    ?string $socket = null,
+    int $flags = 0,
+    bool $enable_env_flags = true
+): \mysqli|false {
+    if ($enable_env_flags) {
+        $flags = $flags | get_environment_connection_flags();
+    }
+
+    return mysqli_real_connect($link, $host, $user, $password, $db, $port, $socket, $flags);
+}
+
+function default_mysqli_connect_ex(): \mysqli|false {
+    return my_mysqli_connect(
+        get_default_host(),
+        get_default_user(),
+        get_default_password(),
+        get_default_database(),
+        get_default_port(),
+        null,
+    );
+}
+function default_mysqli_connect(): \mysqli{
+    $link = default_mysqli_connect_ex();
+    if (!$link) {
+        throw new Error("Cannot connect to default connection");
+    }
+    return $link;
+}
```

Comment:
What exactly is the purpose of these two functions? Why not merge it all into `my_mysqli_connect()`?

Reply:
Part of the idea was to be able to "just" replace the current set of test files which this include instead of the mix of includes currently in place.

But for the ones I've touched I just kinda rewrote them on the way... so for the moment they are very pointless.